contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,978 @@
+     */
+    // rebalance already has re-entrency checks
+    // slither-disable-start reentrancy-no-eth
+    function _addLiquidity() internal gaugeUnstakeAndRestake {
+        uint256 _wethBalance = IERC20(WETH).balanceOf(address(this));
+        uint256 _oethbBalance = IERC20(OETHb).balanceOf(address(this));
+        require(_wethBalance > 0, "Must add some WETH");
+
+        uint160 _currentPrice = getPoolX96Price();
+        /**
+         * Sanity check active trading price is positioned within our desired tick.
+         *
+         * We revert even though price being equal to the lower tick would still
+         * count being within lower tick for the purpose of Sugar.estimateAmount calls
+         */
+        if (
+            _currentPrice <= sqrtRatioX96TickLower ||
+            _currentPrice >= sqrtRatioX96TickHigher
+        ) {
+            revert OutsideExpectedTickRange(getCurrentTradingTick());
+        }
+
+        /**
+         * If estimateAmount1 call fails it could be due to _currentPrice being really
+         * close to a tick and amount1 is a larger number than the sugar helper is able
+         * to compute.
+         *
+         * If token addresses were reversed estimateAmount0 would be required here
+         */
+        uint256 _oethbRequired = helper.estimateAmount1(
+            _wethBalance,
+            address(0), // no need to pass pool address when current price is specified
+            _currentPrice,
+            lowerTick,
+            upperTick
+        );
+
+        if (_oethbRequired > _oethbBalance) {
+            IVault(vaultAddress).mintForStrategy(
+                _oethbRequired - _oethbBalance
+            );
+        }
+
+        uint256 _wethAmountSupplied;
+        uint256 _oethbAmountSupplied;
+        if (tokenId == 0) {
+            (
+                tokenId,
+                ,
+                _wethAmountSupplied,
+                _oethbAmountSupplied
+            ) = positionManager.mint(
+                /** amount0Min & amount1Min are left at 0 because slippage protection is ensured by the
+                 * _checkForExpectedPoolPrice
+                 *›
+                 * Also sqrtPriceX96 is 0 because the pool is already created
+                 * non zero amount attempts to create a new instance of the pool
+                 */
+                INonfungiblePositionManager.MintParams({
+                    token0: WETH,
+                    token1: OETHb,
+                    tickSpacing: tickSpacing,
+                    tickLower: lowerTick,
+                    tickUpper: upperTick,
+                    amount0Desired: _wethBalance,
+                    amount1Desired: _oethbRequired,
+                    amount0Min: 0,
+                    amount1Min: 0,
+                    recipient: address(this),
+                    deadline: block.timestamp,
+                    sqrtPriceX96: 0
+                })
+            );
+        } else {
```

Comment:
⚪️ What if we required token_id to not equal zero right after the mint. This would ensure that we never get into the "we have a token but don't know it" situation.

Reply:
Not sure what you mean. Do you mean add liquidity not having a branching logic and assuming it always has a `tokenId` present?

`What if we required token_id to not equal zero right after the mint`
TokenId is always non zero after the mint (liquidity mint that is)